### ISSUE TITLE
[Legitimate Site Blocked] bytetrade.com bytetrade.io

### ISSUE THREAD

**Issue by pengpeng** [MetaMask/eth-phishing-detect#7351]
Sorry to disturb you all, we hope to make further explanations toward this issue https://github.com/MetaMask/eth-phishing-detect/issues/3624 and remove ‘bytetrade’ related domains from the blacklist.

The maintainer pointed out the reason for blacklisting us in Issue 3624:

The userID is the users secret when importing their account via mnemonic phrase or private key. Please see [https://www.youtube.com/watch?v=hcIdnu86GSA&feature=youtu.be](https://www.youtube.com/watch?v=hcIdnu86GSA&feature=youtu.be)

This warrants the blacklist as per [What reasons do we blacklist a website for?](https://cryptoscamdb.org/faq)

I think that the evaluation method in the video is flawed.
At the 9th second of the video(see below screenshot) shows:
![v1](https://user-images.githubusercontent.com/921131/167434882-7cae546e-5244-423d-9a95-ec488ac315a3.png)
The input box states "Import the mnemonic and account name, separated by spaces"
Mnemonic contains 12 words, and account name also contains 1 word. My understanding is that 13 words need to be entered here, at least it should be more than one word.
![v2](https://user-images.githubusercontent.com/921131/167434978-36ce87fb-16ac-474e-913c-1210271f0fe5.png)
The evaluator pastes a private key which is considered as only one word into the input box.
This input does not meet the "Import the mnemonic and account name, separated by spaces" requirement.
This is the part of the video accusing us:
![v3](https://user-images.githubusercontent.com/921131/167435044-46068cc6-43c3-414f-9502-1f730942b5a9.png)
We include the input in the URL when sending request to the server, "cmd=getUserInviter&userid=6ac....af0e"
We can clearly see that this parameter is userid, and its acquisition method is that “the front-end developer splits the string which is entered in the input box with spaces, and the last one of the array after split is used as userid".

**Why do we need access to the server here?**

Our account system is designed with reference to Bitshares.
In Metamask, entering 12 mnemonics will always get a valid address, although it may not be the user's real address (for example, the user entered the wrong order of the mnemonic)
In Bytetrade, users create orders and make transfers through account name (it can be understood that the blockchain has built-in ens). If the address converted from the mnemonic is inconsistent with the result recorded in the blockchain, the user will be rejected in the following operations. For user experience consideration, the site checks the consistency upon import mnemonics.
If users do not use the website, the developer can directly use the SDK to sign the transaction. If the mnemonic is wrong, the users will only get the notification that the signature is wrong.
As for why not enter the 12 mnemonics directly and why the additional account name is still required. It is mainly used to avoid the problem of not knowing which account to import when a mnemonic phrase holds multiple accounts.

The judging criteria given with this video is [https://cryptoscamdb.org/faq](https://cryptoscamdb.org/faq)

What reasons do we blacklist a website for?
- Impersonating popular crypto services.
- Impersonating popular crypto personalities.
- Sending user-input private keys/secrets to backend services.
- Conducting a giveaway that requires participant's pay an advance fee.
- Preying on, or lying to, investors of a project which can be proved without doubt.
- Pyramid scheme-like language.

My understanding is that the maintainers of eth-phishing-detect think that we are violating the third rule.
I don't think it's appropriate to ignore the input box requirements and unilaterally emphasize whether the evaluation method specified in 3rd rule is triggered. Please take a look at this video: [https://www.youtube.com/watch?v=OsEx4PnFGH0](https://www.youtube.com/watch?v=OsEx4PnFGH0) .
![image](https://user-images.githubusercontent.com/921131/167436796-fda55040-6510-401f-92b0-db2d391ca141.png)
Google also submitted the private key to the server, which also violated the "Sending user-input private keys/secrets to backend services." standard.
I can construct a lot of cases like this. But obviously, Google is not a spam site and we need to provide input content according to the requirements of the input box.

**Will the input content required by Import be misleading to the user?**
I don't think so.
A normal user would first register on the site. Before the registration is completed, he will come to a page like this, and get a prompt to save the mnemonic very well:
![v4](https://user-images.githubusercontent.com/921131/167435703-8d9e1fdc-257f-468c-8335-3a35745467c4.png)
After clicking ‘Save as a file’, the mnemonic format he exported is:

> phone knee surge rotate blood joke depend panel envelope bus leave ice
> testtest2

Users only need to copy the content in the export file to Import input field to complete the import.

It should be pointed out here that:
1. The above picture is a screenshot of the current product instead of 2 years ago when the blacklist happened. But aside from the visual inconsistencies, the interactions on this page are consistent.
2. The exported file format has not changed

I guess the evaluator in the video has not registered for a Bytetrade account.
My understanding is that what the Import dialog asks for has nothing to do with the Ethereum private key.
The mnemonic is encrypted and stored in the browser's localstorage. The user needs to enter a password to decrypt the plaintext. The private key will not be transmitted in any form, and the storage of the private key is safe.

**Let me summarize:**
1. Are we stealing the user's private key?  No
2. Is our website developed well enough? Not good enough
3. Are there potential security risks and further user losses?  As far as I know no. If there is, you can contact me, my email is [pengpeng@bytetrade.io](mailto:pengpeng@bytetrade.io). We can open the case by case analysis. If it is our problem, we will compensate.

**We hope:**
1. We would like to remove the domains [bytetrade.com](http://bytetrade.com/) and [bytetrade.io](http://bytetrade.io/) from the blacklist for the following reasons: we comply with the standard of  [When do we remove Criteria for entries from the blacklist?](https://cryptoscamdb.org/faq)
2. If removal is still not possible, we would like to have an open and transparent discussion about this issue involving the maintainers of MetaMask and eth-phishing-detect.

For two years, this incident has caused us a lot of trouble, but to this day, we still feel that our lack of communication caused the misunderstanding between us and maintainers of MetaMask and eth-phishing-detect. Bytetrade's influence cannot be compared to Metamask, let alone Ethereum. But we are still a serious group of Web3 developers and explorers who want to allow users to own self data. We want this diversity to be treated with respect and fairness.
MetaMask is the superman in the blockchain world, with great power comes great responsibility.
We hope that MetaMask and eth-phishing-detect can revise the previous decision, I can submit a PR accordingly if needed. 

Thank you very much.

Best regards,
Peng Peng

**Comment by pengpeng** [comment]
@devenblake @AlexHerman1 @409H 
May I know the follow-up plan

**Comment by deshvin** [comment]
@409H 
These domains are still in the blocklist as well as blocked by Phishfort. Curious as to your opinion since it looks like some changes have been made on the domain owners end (continued from #3624)

**Comment by 409H** [comment]
@deshvin it looks like the domain content has changed now and not hosting the interface sending secrets to a backend service

**Comment by pengpeng** [comment]
@deshvin

Thank you for your time. For me, this issue is like the tombstone of the project, and I occasionally come to mourn it late at night.

The relevant evaluation methods are inappropriate from the beginning. We never collected private keys/secrets.
After the pain and the struggle, all I can say is, let it be there with its last dignity.

**Comment by deshvin** [comment]
@pengpeng 
I'll handle these domains today, should be resolved within 24 hrs.

**Comment by deshvin** [comment]
In addition, these domains were blocked by a 3rd party list that we absorb. I've reached out to them to have the domains removed from their blocklist as well.

The phishing warning should no longer appear within 24 hours.

**Comment by pengpeng** [comment]
@deshvin Thank you
